databricks/koalas/groupby.py

```diff
@@ -340,6 +344,102 @@ def any(self):
             lambda col: F.max(F.coalesce(col.cast('boolean'), F.lit(False))),
             only_numeric=False)
 
+    # TODO: Series support is not implemented yet.
+    def transform(self, func):
+        """
+        Apply function column-by-column to the GroupBy object.
+
+        The function passed to `transform` must take a Series as its first
+        argument and return a Series. The given function is executed for
+        each series in each grouped data.
+
+        While `transform` is a very flexible method, its downside is that
+        using it can be quite a bit slower than using more specific methods
+        like `agg` or `transform`. Koalas offers a wide range of method that will
+        be much faster than using `transform` for their specific purposes, so try to
+        use them before reaching for `transform`.
+
+        .. note:: unlike pandas, it is required for ``func`` to specify return type hint.
+
+        .. note:: the series within ``func`` is actually a pandas dataframe. Therefore,
+            any pandas APIs within this function is allowed.
+
+        Parameters
+        ----------
+        func : callable
+            A callable that takes a Series as its first argument, and
+            returns a Series.
+
+        Returns
+        -------
+        applied : DataFrame
+
+        See Also
+        --------
+        aggregate : Apply aggregate function to the GroupBy object.
+        Series.apply : Apply a function to a Series.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'A': [0, 0, 1],
+        ...                    'B': [1, 2, 3],
+        ...                    'C': [4, 6, 5]}, columns=['A', 'B', 'C'])
+
+        >>> g = df.groupby('A')
+
+        Notice that ``g`` has two groups, ``0`` and ``1``.
+        Calling `transform` in various ways, we can get different grouping results:
+        Below the functions passed to `transform` takes a Series as
+        its argument and returns a Series. `transform` applies the function on each series
+        in each grouped data, and combine them into a new DataFrame:
+
+        >>> def convert_to_string(x) -> ks.Series[str]:
+        ...    return x.apply("a string {}".format)
+        >>> g.transform(convert_to_string)  # doctest: +NORMALIZE_WHITESPACE
+                    A           B           C
+        0  a string 0  a string 1  a string 4
+        1  a string 0  a string 2  a string 6
+        2  a string 1  a string 3  a string 5
```

Comment:
Seems like pandas doesn't apply for the grouping columns:

```py
>>> df = pd.DataFrame({'A': [0, 0, 1], 'B': [1, 2, 3], 'C': [4, 6, 5]})
>>> g = df.groupby('A')

>>> def convert_to_string(x):
...   return x.apply("a string {}".format)
...
>>> g.transform(convert_to_string)
            B           C
0  a string 1  a string 4
1  a string 2  a string 6
2  a string 3  a string 5
```

Reply:
Ah.. Yes. I completely forgot to exclude this column. Will update it soon.